### DISCUSSION TITLE
Help with LSP autocompletion

### DISCUSSION THREAD

**Discussion by itmecho** [mantinedev/mantine#224] [Q&A] [answered]
Loving the project but I can't seem to get autocomplete to work in either `neovim` or `code`. I'm asking here rather than submitting an issue because I'm pretty sure it's going to be something stupid about my environment rather than an issue with mantine!

### Neovim
![image](https://user-images.githubusercontent.com/8384983/130763896-dbbbd552-ce8b-4b21-a501-b9249ad60c24.png)

### Code
![image](https://user-images.githubusercontent.com/8384983/130763950-e9fa876e-d260-4075-b020-7587bdc90b13.png)

### `tsconfig.json`
```json
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "baseUrl": ".",
    "paths": {
      "@itmecho/*": "./*"
    }
  },
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

**Comment by rtivital** [comment, marked as answer]
Hi @itmecho, please make sure that you use latest version of Mantine or at least latest patch of your minor version. If it does not help, please see if you can replicate this issue in next-template – https://github.com/mantinedev/mantine-next-template

**Reply by itmecho** [reply]
Thanks @rtivital! I think I'm on the latest version already:
```json
{
  "name": "itmecho.com",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@mantine/core": "^2.3.2",
    "@mantine/hooks": "^2.3.2",
    "@mantine/prism": "^2.3.2",
    "next": "11.1.0",
    "prism-react-renderer": "^1.2.1",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-jss": "^10.7.1"
  },
  "devDependencies": {
    "@types/react": "17.0.19",
    "eslint": "7.32.0",
    "eslint-config-next": "11.1.0",
    "prettier": "^2.3.2",
    "typescript": "4.3.5"
  }
}
```

I'll try with that template now!

**Reply by itmecho** [reply]
Hmm still not working in the template project =\ 

![image](https://user-images.githubusercontent.com/8384983/130770730-303b21ca-2157-4130-a83a-f9c997c5eb70.png)

It knows the type of things if I manually import them though!

![image](https://user-images.githubusercontent.com/8384983/130770819-bc1616e1-b3ea-43ae-9b5f-17cf516338a6.png)

**Reply by itmecho** [reply]
And it properly auto completes things that are already imported:

![image](https://user-images.githubusercontent.com/8384983/130770901-98c5e02f-7dfa-4cc9-adf5-b83d81958202.png)

**Reply by rtivital** [reply]
Well, that means that is works as expected, I do not know your workflow. When I write Button it autoimports it and has correct types (vs code).

**Reply by itmecho** [reply]
Yea it must be something to do with my system but I'm not really sure where to look =\ For reference, vanilla vs code with no plugins also doesn't work for me.

**Reply by itmecho** [reply]
Ah OK, I've found the problem! I'm using `pnpm`. If I do `rm -Rf node_modules && yarn install` it works!

Interestingly, if I install `@material-ui/core` using `pnpm` in the `mantine-next-template` repo, that does show up in the autocomplete. I guess it's doing something funky with symlinks

**Reply by itmecho** [reply]
I think this might be an issue with the `typescript-language-server`. I've opened an issue there to see if anyone can help!

https://github.com/typescript-language-server/typescript-language-server/issues/254

**Reply by itmecho** [reply]
Even more interesting is that it works with `@mantine/hooks` :confused:
